src/platform/packages/shared/kbn-discover-contextual-components/src/data_types/logs/components/summary_column/utils.tsx

```diff
@@ -147,54 +139,47 @@ const getResourceBadgeIcon = (
   }
 };
 
-const getFormattedValue = (
-  name: keyof TraceFields,
-  rawValue: string,
-  dataView: DataView
-): string => {
+const getFormattedValue = (name: FieldKey, rawValue: unknown, dataView: DataView): string => {
   const field = dataView.getFieldByName(name);
   const formatter = field && dataView.getFormatterForField(field);
 
   if (formatter) {
-    return formatter.convert(rawValue);
-  } else if (DURATION_FIELDS.includes(name)) {
-    return `${rawValue}µs`;
+    return formatter.convert(rawValue, 'html', { field });
   }
 
   return `${rawValue}`;
 };
 
-export const isTraceDocument = (row: DataTableRecord): row is TraceDocument =>
-  getFieldValue(row, DATASTREAM_TYPE_FIELD) === 'traces';
+const isTracesIndex = testPatternAgainstAllowedList(['traces']);
 
-export const createTraceFields = (
-  row: TraceDocument,
-  dataView: DataView,
-  core: CoreStart,
-  share?: SharePluginStart
-): ResourceFieldDescriptor[] => {
-  const traceFields = getUnformattedFields(row, TRACE_FIELDS);
-  const availableFields = getAvailableTraceFields(traceFields);
-
-  return availableFields.map((name) => ({
-    name,
-    value: getFormattedValue(name, traceFields[name], dataView),
-    ResourceBadge: getResourceBadgeComponent(name, core, share),
-    Icon: getResourceBadgeIcon(name, traceFields),
-  }));
-};
+export const isTraceDocument = (row: DataTableRecord): row is TraceDocument =>
+  getFieldValue(row, DATASTREAM_TYPE_FIELD) === 'traces' ||
+  isTracesIndex(getFieldValue(row, INDEX_FIELD) as string);
+
+interface ResourceFieldsProps {
+  row: DataTableRecord;
+  fields: readonly FieldKey[];
+  getAvailableFields: (doc: Readonly<Record<FieldKey, FieldValue>>) => FieldKey[];
+  dataView: DataView;
+  core: CoreStart;
+  share?: SharePluginStart;
+}
 
-export const createResourceFields = (
-  row: DataTableRecord,
-  core: CoreStart,
-  share?: SharePluginStart
-): ResourceFieldDescriptor[] => {
-  const resourceDoc = getUnformattedFields(row as LogDocument, RESOURCE_FIELDS);
-  const availableResourceFields = getAvailableResourceFields(resourceDoc);
+export const createResourceFields = ({
+  row,
+  fields,
+  getAvailableFields,
+  dataView,
+  core,
+  share,
+}: ResourceFieldsProps): ResourceFieldDescriptor[] => {
+  const resourceDoc = getUnformattedFields(row, fields);
+  const availableResourceFields = getAvailableFields(resourceDoc);
 
   return availableResourceFields.map((name) => ({
     name,
-    value: resourceDoc[name],
+    rawValue: resourceDoc[name],
+    value: getFormattedValue(name, resourceDoc[name], dataView),
```

Comment:
```suggestion
    value: formatFieldValue(
      resourceDoc[name],
      row.raw,
      fieldFormats,
      dataView,
      dataView.getFieldByName(name),
      'html'
    ),
```
Since there's no longer any special handling going on here, I'd definitely recommend dropping the `getFormattedValue` function and using `formatFieldValue` since it's the standard util and will produce the best output.

Reply:
I've made the change to make use of `formatFieldValue`, but as a note, there's a soundness hole with it in that in some circumstances, it doesn't actually return a `string` value. This is due to `FieldFormat.convert` not actually turning a value into `string` if there is no formatter present for it, instead it simply casts it to string. As such, I put TODOs on the places I've changed to using `formatFieldValue` to remove the string guards once the method no longer has a soundness hole.

Reply:
Thanks for updating it 👍 And interesting, TIL. I suppose that would be an issue regardless for any use of `getFormatterForField`, but I don't think it's an actual scenario. It looks like `converter` never returns undefined and we we just have a redundant null check there. We should be able to drop it and get rid of `value as string`. Our team owns that code so we can take a look at it as a followup and update here too.